kork/kork-web/src/main/java/com/netflix/spinnaker/config/yaml/YamlAutoConfiguration.java

```diff
@@ -0,0 +1,60 @@
+package com.netflix.spinnaker.config.yaml;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+
+@AutoConfiguration
+@EnableConfigurationProperties(YamlParserProperties.class)
+@Slf4j
+public class YamlAutoConfiguration {
```

Comment:
javadoc please.  The commit message has some good info that would help here.

All of this auto-config and conditional stuff seems pretty fancy.  Do we really need it?  Seems like changing the config properties from int to Integer, or somehow making it clear in the code whether they've been specified or not would be simpler.  Plus also, how/why would a Yaml bean exist in the spring context?

Reply:
Plus also, does anything use this Yaml bean, or does that happen in a future commit?